src/main/java/com/mate/academy/demo/service/BookServiceImpl.java

```diff
@@ -28,6 +29,19 @@ public BookDto findById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Book not found by id " + id)));
     }
 
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto bookDto, Long id) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book not found by id " + id));
+        Book book = bookMapper.toModelComplete(bookDto, id);
```

Comment:
create in mapper
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`

```suggestion
        bookMapper.updateBookFromDto(bookDto, book);
```